test/processors.jl

```diff
@@ -40,23 +48,54 @@ end
             @test Dagger.iscompatible(proc, opts, unknown_func, us, 1, us, 2.0)
         end
     end
+    @testset "Function/argument compatability of DaggerKAProc" begin
+        unknown_func = () -> nothing
+        tp = DaggerKAProc(1)
+        op = get_parent(tp)
+        opts = ThunkOptions()
+        us = UnknownStruct()
+        for proc in (op, tp)
+            @test Dagger.iscompatible_func(proc, opts, unknown_func)
+            @test Dagger.iscompatible_arg(proc, opts, us)
+            @test Dagger.iscompatible(proc, opts, unknown_func, us, 1)
+        end
+    value(a) = a
+    @kernel function add(A, B, C)
+        I = @index(Global)
+        C[I] = A[I] + B[I]
+    end
+    A = delayed(value)(ones(512,512))
+    B = delayed(value)(3*ones(512,512))
+    C = delayed(value)(zeros(512,512))
+    S = delayed(add)(A,B,C)
+    @test all(collect(S) .== 4.0)
```

Comment:
We should be testing the value of `C` as well; the fact that `S` has a value is somewhat irrelevant, since the point of the kernel is to do an elementwise sum of `A .+ B` and store it in `C`.

Reply:
Yes, I had tried that but since the kernel function won't be accessible if all the variables are not initialised. So, if I am going to pass values of (A,B,C) to the kernel function, it needs to be stored somewhere. So, the ``S`` object stores the value of ``C``.

Reply:
Try making these modifications to your example:
```
function addk(p,q,r)
  r .= p .+ q # Use broadcast to assign into the array `r`, instead of reassigning `r`
end

R = zeros(512,512)
r = delayed(value)(R)
```

And then:
```
collect(j) # We don't care about the return value of `j`
all(R .== 4.0)
```

Reply:
ok

Reply:
its `` all(r .== 4.0)``